src/RequestExecutor/HttpRequestExecutor.ts

```diff
@@ -190,6 +197,15 @@ export class HttpRequestExecutor implements RequestExecutor {
   }
 
   private getRequestAgent(options: Request) {
+    if (
+      this.proxyDomains &&
+      this.proxyDomains.some((domain) =>
```

Comment:
Looks like there should invert
```suggestion
      !this.proxyDomains.some((domain) =>
```
if proxy domains defined and url does not match any of provided domains then we should not send request through proxy